src/halmos/__main__.py

```diff
@@ -1089,7 +1114,7 @@ def parse_build_out(args: argparse.Namespace) -> Dict:
     return result
 
 
-def main() -> int:
+def main(argv=None, test_results_map=None) -> int:
```

Comment:
it seems a little odd to pass a "write argument" to main, but ok 🙃 could use a docstring explaining that it's basically for testing, to essentially avoid going through file IO right?

Reply:
i wasn't happy with this either, but did so to avoid unnecessary `asdict` conversion for test results in normal execution.  i changed the code to return the test results.